Maman11/Q1/my_add.c

```diff
@@ -1,25 +1,10 @@
 #include <stdio.h>
 #include <stdlib.h>
+#define DATA_SIZE 33
 
-int main() {
-    int num1, num2, binary1, binary2;
-    printf("Enter a number: ");
-    scanf("%d", &num1); // Reads an integer from the user
-
-    printf("Enter a second number: ");
-    scanf("%d", &num2); // Reads an integer from the user
-
-    printf("\nYou entered : %d and %d\n\n", num1, num2);
-
-    binary1 = decimalToBinary(num1);
-    binary2 = decimalToBinary(num2);
-    printf("\nThe sum of %d and %d is: %d or %d in decimal\n", num1, num2, addBinary(binary1, binary2), num1+num2);
-    return 0;
-}
-
-int decimalToBinary(int num){
+unsigned int decimalToBinary(int num){
```

Comment:
This function is wrong.
Converting a number to a string and then to a number is not efficient both in memory and in compute.
Try to convert a number to string and then to return the string.

Think about the number 1025 its binery value is 10000000001 this value is bigger then int.MaxValue that is 2147483647

Thus your program will not work

Reply:
I have to convert it to unsigned int one way or another since it's required that my_add returns an unsigned int. Does it matter if I do in the my_add or decimalToBinary?

Reply:
I think you got this a bit wrong. 

my_add - A function that only do a binary addition 

Also you should have a function that converts a number to a string in a binary representation. 
So in other words you have a function that gets a number (type is unsigned int because the program uses unsigned int for numbers in this case)
And returns the output that is the given number`s string in a binary representation. 

in other words 
char* numToBinStr(unsigned int)